ocaml/typing/persistent_env.ml

```diff
@@ -107,14 +125,16 @@ let empty () = {
 }
 
 let clear penv =
-  let {
+  let [@warning "+missing-record-field-pattern"] {
```

Comment:
It's _extremely_ valuable here. I don't know any other way to prod someone to add code here if they add a new field. (It's too bad there isn't an “opposite of `_`” to force a record pattern to _be_ exhaustive.) If it's a wart that makes people want to enable the warning more broadly, that seems win/win? Does seem worth adding a comment.

Reply:
Taking it out, but it still makes me nervous. It strikes me as a place where it's both very easy to make this error (who would think to check all pattern matches on an environment record?) _and_ very easy for the bug to remain undetected until it has subtle effects later as global state silently gets threaded along. I did add a comment to the record type imploring my future self to handle the footgun carefully.

Reply:
(On reflection, we can actually just _delete the whole `clear` function_. It's only called by `Env`, which keeps a reference, and it should just re-initialize the reference to `Persistent_env.empty ()`. I'll make a PR next time I think about it, which will be soon because the comment I added will annoy me.)